src/main/java/intefaceAdapters/createListing/ListingRepo.java

```diff
@@ -0,0 +1,95 @@
+package intefaceAdapters.createListing;
+
+import useCases.createListing.CreateListingDsGateway;
+import useCases.createListing.CreateListingDsRequestModel;
+import useCases.createListing.CreateListingRequestModel;
+import useCases.userRegister.UserRegisterDsRequestModel;
+
+import java.io.*;
+import java.time.LocalDateTime;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class ListingRepo implements CreateListingDsGateway {
+    private File csvFile;
+    private final Map<String, Integer> headers = new LinkedHashMap<>();
+    private final Map<String, CreateListingDsRequestModel> listingsMap = new HashMap<>();
+
+    public ListingRepo (String csvPath) throws IOException {
```

Comment:
Instead of using a map, you could use a loop from 0 to 11 to store the headers. It should get rid of hardcoding that we are doing.